.github/actions/deploy-cluster-run-tests/action.yml

```diff
@@ -0,0 +1,20 @@
+name: "Deploy Kind cluster and run tests"
+description: "Restores k8s from the cache previously built, installs kind, builds and runs e2e tests and exports logs and reports"
```

Comment:
this does not restore from the cache ... it downloads an artifact that we uploaded previously